plugins/source/hackernews/resources/services/items/items_fetch.go

```diff
@@ -49,12 +57,23 @@ func fetchItems(ctx context.Context, meta schema.ClientMeta, _ *schema.Resource,
 			Str("table", tableName).Str("client_id", c.ID()).Msg("Found previous cursor with value " + strconv.Itoa(cursor))
 	}
 
-	// find the max item ID from the Hacker News API
-	maxID, err := c.HackerNews.MaxItemID(ctx)
-	if err != nil {
-		return err
+	// we allow the user to specify a start time for posts, so we need to find the first post after that time
+	if c.Spec.StartTime != "" {
+		startTime, _ := time.Parse(time.RFC3339, c.Spec.StartTime)
+		c.Logger().Info().Msgf("Finding first post after %s", startTime)
+		startItemID, err := findFirstPostAfter(ctx, c, startTime, maxID)
+		if err != nil {
+			return err
+		}
+		c.Logger().Info().Msgf("Found first post after %s with id %d", startTime, startItemID)
```

Comment:
more structured logging needed :) also in line 63

Reply:
done 👍